src/core/renderer.js

```diff
@@ -7,6 +9,19 @@ export class Renderer {
     // Abstract method
   }
 
+  static shouldRefreshChildFrameWithMorphing(parentFrame, currentFrame, newFrame) {
+    return currentFrame instanceof FrameElement &&
+      // newFrame cannot yet be an instance of FrameElement because custom
+      // elements don't get initialized until they're attached to the DOM, so
+      // test its Element#nodeName instead
+      newFrame instanceof Element && newFrame.nodeName === "TURBO-FRAME" &&
+      currentFrame.shouldReloadWithMorph &&
+      currentFrame.id === newFrame.id &&
+      (!newFrame.getAttribute("src") || urlsAreEqual(currentFrame.src, newFrame.getAttribute("src"))) &&
```

Comment:
Why the "urls are equal" condition?

Reply:
Asking because, if we could get away without comparing URL and IDs, then we could just rely on making the check a single-param function (currentFrame).

Reply:
This check is still addressing a valid concern:
`<turbo-frame id="a" src="/a">` => `<turbo-frame id="a">` Reload!
`<turbo-frame id="a" src="/a">` => `<turbo-frame id="a" src="/a">` Reload!
`<turbo-frame id="a" src="/a">` => `<turbo-frame id="a" src="/b">` Do NOT reload. Instead, morph as usual so that the `src` attribute gets mutated, and thus the turbo-frame loads the new `/b` page, rather than being stuck on `/a`.

Maybe there's a better way to do this, or a better place to do it in, or both! But this is what made sense to me.